core/jazz_t-vault/components/util.js

```diff
@@ -0,0 +1,387 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+const request = require("request");
+const logger = require('./logger.js');
+
+function getVaultToken(configData) {
+  return new Promise((resolve, reject) => {
+    let payload = {
+      uri: `${configData.T_VAULT_API}${configData.T_VAULT_LOGIN_API}`,
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "Accept": "application/json"
+      },
+      json: {
+        "username": configData.T_VAULT_USERNAME,
+        "password": configData.T_VAULT_PASSWORD
+      }
+    };
+
+    logger.debug("login payload : " + JSON.stringify(payload));
+    request(payload, function (error, response, body) {
+      logger.debug("login response : " + JSON.stringify(response));
+      if (response.statusCode && (response.statusCode === 200 || response.statusCode === 201) && body && body.client_token) {
```

Comment:
Why is response code 200 or 201 for the same API?

Reply:
Done